src/PrestaShopBundle/Resources/config/services/adapter/admin.yml

```diff
@@ -8,5 +8,10 @@ services:
             - "@session"
         decorates: prestashop.core.admin.page_preference_interface
         public: false
+
     prestashop.adapter.admin.controller.category:
         class: PrestaShop\PrestaShop\Adapter\Category\AdminCategoryControllerWrapper
+
+    prestashop.adapter.legacy_db:
+        class: 'Db'
+        factory: ['Db', 'getInstance']
```

Comment:
class `PrestaShop\PrestaShop\Core\Addon\Theme\ThemeManagerBuilder` even though it's in `Core` it heavily depends on legacy `Db` so i had to do this :cry: :cry:  :cry:

Reply:
😭 
Can we add a `@todo` here ? so that we fix this later ?

I know I bother you a lot with `@todo` statements
But it's really helpful when you want to look for technical debt, 1 way to find some is to look for them.

Reply:
i think that `ThemeManagerBuilder` would be a better place for todo as it has to be refactored not to depend on legacy classes, wdyt?

Reply:
actually i got this idea: what if we make `Adapter` and other legacy services deprecated when declaring? For example:

```yaml
    prestashop.adapter.legacy_db:
        class: 'Db'
        factory: ['Db', 'getInstance']
        deprecated: ~ # we could even add a deprecation message here
```
Doing that, would be another way of telling developers not to depend on them and in addition to that, all deprecation notices would be visible on profiler, wdyt?

Reply:
The idea is interesting but it is not really deprecated as it's still in use 🤔

Reply:
at least we get notices in profiler when using it, wdyt?

Reply:
Good point 👍